src/UKHO.ERPFacade.API/Helpers/LicenceUpdatedSapMessageBuilder.cs

```diff
@@ -0,0 +1,134 @@
+            {
+                _logger.LogError(EventIds.LicenceUpdatedSapXmlTemplateNotFound.ToEventId(), "The licence updated SAP message xml template does not exist.");
+                throw new FileNotFoundException();
+            }
+
+            _logger.LogInformation(EventIds.CreatingLicenceUpdatedSapPayload.ToEventId(), "Creating licence updated SAP Payload.");
+
+            XmlDocument soapXml = _xmlHelper.CreateXmlDocument(sapXmlTemplatePath);
+
+            var sapRecordOfSalePayLoad = SapXmlPayloadCreation(eventData);
+
+            string xml = _xmlHelper.CreateRecordOfSaleSapXmlPayLoad(sapRecordOfSalePayLoad);
+
+            string sapXml = RemoveNullFields(xml.Replace(ImOrderNameSpace, ""));
+            soapXml.SelectSingleNode(XpathZAddsRos).InnerXml = sapXml.SetXmlClosingTags();
+
+            _logger.LogInformation(EventIds.CreatedLicenceUpdatedSapPayload.ToEventId(), "Licence updated SAP payload created.");
+
+            return soapXml;
+        }
+
+        private SapRecordOfSalePayLoad SapXmlPayloadCreation(RecordOfSaleEventPayLoad eventData)
+        {
+            var sapPayload = new SapRecordOfSalePayLoad
+            {
+                CorrelationId = eventData.Data.CorrelationId,
+                ServiceType = eventData.Data.Licence.ProductType,
+                LicTransaction = eventData.Data.Licence.TransactionType,
+                SoldToAcc = eventData.Data.Licence.DistributorCustomerNumber,
+                LicenseEacc = eventData.Data.Licence.ShippingCoNumber,
+                LicenceNumber = eventData.Data.Licence.SapId,
+                VesselName = eventData.Data.Licence.VesselName,
+                IMONumber = eventData.Data.Licence.ImoNumber,
+                CallSign = eventData.Data.Licence.CallSign,
+                ShoreBased = eventData.Data.Licence.ShoreBased,
+                FleetName = eventData.Data.Licence.FleetName,
+                Users = eventData.Data.Licence.NumberLicenceUsers,
+                EndUserId = eventData.Data.Licence.LicenceId,
+                ECDISMANUF = eventData.Data.Licence.Upn,
+                OrderNumber = eventData.Data.Licence.TransactionType == TransactionType ? "" : eventData.Data.Licence.OrderNumber,
+                StartDate = eventData.Data.Licence.TransactionType == TransactionType ? "" : eventData.Data.Licence.OrderDate,
+                PurachaseOrder = eventData.Data.Licence.TransactionType == TransactionType ? "" : eventData.Data.Licence.PoRef,
+                EndDate = eventData.Data.Licence.TransactionType == TransactionType ? "" : eventData.Data.Licence.HoldingsExpiryDate,
+                LicenceType = eventData.Data.Licence.TransactionType == TransactionType ? "" : eventData.Data.Licence.LicenceType,
+                LicenceDuration = eventData.Data.Licence.TransactionType == TransactionType ? null : eventData.Data.Licence.LicenceDuration
+            };
+
+            sapPayload.PROD = new PROD()
```

Comment:
Any reason we are hardcoding these values and not adding what we receive from EES?

Reply:
Hi @Vishal14583, As we are not receiving any value for unit of sale  so think that  and SAP also expecting the empty value unit of sale order so we explicitly added empty value  there, even if we received any values for unit of sale order still we can send the empty value  to SAP.